blog-investigation/src/shared/store/UserSettingsProvider.jsx

```diff
@@ -0,0 +1,24 @@
+import React, { useReducer, useContext, useMemo } from 'react'
+import { UserSettingsContext } from './context'
+import userSettingsReducer, { userSettingsInitialState } from './reducer'
+
+const UserSettingsProvider = ({children}) => {
+    const [state, dispatch] = useReducer(userSettingsReducer, userSettingsInitialState) 
+
+    const userSettingsContextValue = useMemo(() => {
+        return {
+            ...state,
```

Comment:
causes rerenders, put theme, language, isLoggedIn as  keys

Reply:
I have state as key, should I add its props again?

Reply:
yes, please add. spread operator `...` generates a new reference